CalPatRec/src/PhiClusterFinder_module.cc

```diff
@@ -0,0 +1,566 @@
+      }
+    }
+    if(_debug > 0 and tccol.size() > 0){
+      int newsize = tccol1->size();
+      int oldsize = tccol.size();
+      int eventno = _iev;
+      printf("Output tccol size = %i Input tccol size =  %i event = %i\n",newsize,oldsize,eventno);
+    }
+
+    TimeClusterCollection& tccolnew(*tccol1);
+    // if(newsize == 2 and _debug>2) std::cout<<"Event with two phi clusters = "<<_iev<<std::endl;
+
+    // Check for calo clusters
+    if(_useCC and tccol.size() > 0){
+      addCaloClusters(tccolnew,tccol);
+    }
+    // Save the output time cluster collection for the diagnostics
+    _data._tccolnew = tccol1.get();
+    // if(_debug>0) std::cout<<"Diag data size = "<<_data._tccolnew->size()<<" old = "<<_data._tccol->size()<<std::endl;
+    if (_diag > 0) {
+      _hmanager->fillHistograms(&_data);
+    }
+    // if (tccol1->size() == 0 and tccol.size()>0)
+    // std::cout<<"tccol new size = "<<tccol1->size()<<" tccol size = "<<tccol.size()<<" event = "<<_iev<<std::endl;
+    event.put(std::move(tccol1));
+  }
+
+//-----------------------------------------------------------------------------
+// process input timecluster to find the cluster of hits separated in phi
+//-----------------------------------------------------------------------------
+  void PhiClusterFinder::findClusters(TimeClusterCollection& tccol1, const std::vector<StrawHitIndex>& ordchcol) {
+    int nh = ordchcol.size();
+    // Counts the no. of hits used to form the clusters
+    int countusedhit(0);
+    // Counts the no. of clusters
+    int counter(0);
+    // Count the no. of hits in a cluster when it is an event with two phi clusters
+    int count1(0),count2(0);
+    float dphi(0);
+    // Mark true if a hit is used to form a cluster
+    std::vector<bool> usedhit(ordchcol.size(),false);
+
+    if (nh > (_minnsh/2)) {
+      // Check if all the hits are used and keep trying to find peaks until < 5 hits are left in the collection
+      while(nh-countusedhit > (_minnsh/2)) {
+        int   finalcount(0);
+        float phi1(0);
+        _hist1->Reset("ICSEM");
+
+        if (_debug > 1) printf("Cluster : %2i\n",counter);
+        // Fill the phi histogram with all the unused hits
+        for(int i=0; i<nh; i++){
+          int ind = ordchcol[i];
+          const mu2e::ComboHit* ch = &_chcol->at(ind);
+          if(usedhit[i]==true) continue;
+          Double_t phi = ch->phi();
+          // change the range to [0,2pi]
+          if(phi < 0) phi += 2*M_PI;
+          _hist1->Fill(phi,ch->nStrawHits());
+          // if(_debug>3) std::cout<<"Phi "<<phi<<" ch phi "<< phi << "n straw hits = "<<ch->nStrawHits()<<std::endl;
+        }
+
+        // Find the min and max phi around the highest phi bin in the phi spectrum
+        float cluphimin(0),cluphimax(0);
+        clusterminmax(cluphimin, cluphimax);
+        if (_debug > 2) printf("Min phi = %f Max phi = %f \n",cluphimin,cluphimax);
+        float bin = _hist1->GetBinWidth(0);
```

Comment:
this is a constant as the binning is fixed. It can be moved outside the loop.

Reply:
This is outside the loop.

Reply:
I meant outside of the ‘while’ loop